### PR TITLE
fix(secu): prevent from sql injections in host page

### PR DESCRIPTION
**Pull request by sc979**
# Pull Request Template

## Description

- avoid SQL injection in monitoring host page
- clean code

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [x] 18.10.x
- [x] 19.04.x
- [x] 19.10.x (master)

<h2> How this pull request can be tested ? </h2>

please contact me

#### Community contributors & Centreon team

- [x] I followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

#### Centreon team only

- [ ] I have made sure that the **unit tests** related to the story are successful.
- [ ] I have made sure that **unit tests cover 80%** of the code written for the story.
- [x] I have made sure that **acceptance tests** related to the story are successful (**local and CI**)

### CODE REVIEW

File: www/include/monitoring/status/Hosts/xml/hostXML.php [hunk 1]
```diff
@@ -192,66 +199,85 @@
...
+        // Not SQL portable
+        $rq1 .= " ORDER BY IFNULL(inet_aton(h.address), h.address) " . $order . ",h.name";
         break;
     case 'plugin_output':
-        $rq1 .= " ORDER BY h.output " . $order . ",h.name ";
+        $rq1 .= " ORDER BY h.output " . $order . ",h.name";
         break;
     case 'criticality_id':
-        $rq1 .= " ORDER BY isnull $order, criticality $order, h.name ";
+        $rq1 .= " ORDER BY ISNULL " . $order . ", criticality " . $order . ", h.name";
```
Comment (loiclau) [inline www/include/monitoring/status/Hosts/xml/hostXML.php]: can delete this line, like to default case

File: www/include/monitoring/status/Hosts/xml/hostXML.php [hunk 2]
```diff
@@ -192,66 +199,85 @@
...
         break;
     default:
-        $rq1 .= " ORDER BY isnull $order, criticality $order, h.name ";
+        $rq1 .= " ORDER BY ISNULL " . $order . ", criticality " . $order . ", h.name";
```
Comment (sc979) [inline www/include/monitoring/status/Hosts/xml/hostXML.php]:
```suggestion
        $rq1 .= " ORDER BY isnull " . $order . ", criticality " . $order . ", h.name";
```

Comment (sc979) [reply www/include/monitoring/status/Hosts/xml/hostXML.php]: isnull is an alias

File: www/include/monitoring/status/Hosts/xml/hostXML.php (hunk 1 above)
Comment (sc979) [inline www/include/monitoring/status/Hosts/xml/hostXML.php]:
```suggestion
```

File: www/include/monitoring/status/Hosts/xml/hostXML.php [hunk 3]
```diff
@@ -241,7 +241,6 @@
         $rq1 .= " ORDER BY h.output " . $order . ",h.name";
         break;
     case 'criticality_id':
-        $rq1 .= " ORDER BY ISNULL " . $order . ", criticality " . $order . ", h.name";
         break;
```
Comment (loiclau) [inline www/include/monitoring/status/Hosts/xml/hostXML.php]: this one too

File: www/include/monitoring/status/Hosts/xml/hostXML.php [hunk 4]
```diff
@@ -192,66 +199,84 @@
...
+        // Not SQL portable
+        $rq1 .= " ORDER BY IFNULL(inet_aton(h.address), h.address) " . $order . ",h.name";
         break;
     case 'plugin_output':
-        $rq1 .= " ORDER BY h.output " . $order . ",h.name ";
+        $rq1 .= " ORDER BY h.output " . $order . ",h.name";
         break;
     case 'criticality_id':
-        $rq1 .= " ORDER BY isnull $order, criticality $order, h.name ";
         break;
```
Comment (loiclau) [inline www/include/monitoring/status/Hosts/xml/hostXML.php]:
```suggestion
```

File: www/include/monitoring/status/Hosts/xml/hostXML.php [hunk 5]
```diff
@@ -192,66 +199,83 @@
...
-        # Not SQL portable
-        $rq1 .= " ORDER BY IFNULL(inet_aton(h.address), h.address) " . $order . ",h.name ";
+        // Not SQL portable
+        $rq1 .= " ORDER BY IFNULL(inet_aton(h.address), h.address) " . $order . ",h.name";
         break;
     case 'plugin_output':
-        $rq1 .= " ORDER BY h.output " . $order . ",h.name ";
+        $rq1 .= " ORDER BY h.output " . $order . ",h.name";
         break;
     case 'criticality_id':
```
Comment (sc979) [inline www/include/monitoring/status/Hosts/xml/hostXML.php]: as 'criticality_id' and default case will add the same partial string to the request, we let it empty

**Pull request merged**